packages/rocketchat-integrations/server/processWebhookMessage.js

```diff
@@ -4,7 +4,15 @@ function retrieveRoomInfo({ currentUserId, channel, ignoreEmpty=false }) {
 		throw new Meteor.Error('invalid-channel');
 	}
 
-	if (room && room.t === 'c') {
+	//Since the room object can be empty, return out now since we depend on it to be an object below
+	if (!room) {
+		return room;
+	}
+
+	//Check if the user already has a Subscription or not, this avoids this issue: https://github.com/RocketChat/Rocket.Chat/issues/5477
+	const sub = RocketChat.models.Subscriptions.findOneByRoomIdAndUserId(room._id, currentUserId);
```

Comment:
It's better to test by `room.t` before to prevent do the find for all type of rooms